### ISSUE TITLE
Contextual stem directions

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#1959]
This is a good backburner issue, although if someone wants to work on it now, I am not stopping you :-)

I created a backburner label for issues below the "low proiority" level.  Issues with this label should be set to closed when they are created, even though they are not resolved.  When a backburner issue has been resolved, the backburner label should be removed.  This will allow searching for all backburner issues which are closed but technically open and should be dealt with some time in the future (i.e., when the open issues list approaches 0...).

-------------------------

Here is an example where contextual stem directions are needed:

<img width="260" alt="Screen Shot 2021-01-16 at 6 54 13 AM" src="https://user-images.githubusercontent.com/3487289/104815133-a7304900-57c7-11eb-86ab-a09496419fb4.png">

Notice that the middle beamed group of notes are all equally centered around the middle line of the staff.  Verovio has to make a decision about whether  to place the stems up or stems down.  The general rule is that stems prefer to point in the direction where there is a greater distance to the outside staff lines.  In this case there is a tie between up and down.  When there is such a tie, verovio chooses the downward direction for such stems, which is the standard typesetting rule:

<img width="309" alt="Screen Shot 2021-01-16 at 7 06 56 AM" src="https://user-images.githubusercontent.com/3487289/104815441-6d604200-57c9-11eb-81d8-1215e95b550e.png">

<details>
<summary>
Click to view MEI data for above example.
</summary>

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="https://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
 <meiHead>
  <fileDesc>
   <titleStmt>
    <title />
   </titleStmt>
   <pubStmt />
  </fileDesc>
  <encodingDesc>
   <appInfo>
    <application isodate="2021-01-16T07:07:35" version="3.1.0-dev-0361cac-dirty">
     <name>Verovio</name>
     <p>Transcoded from Humdrum</p>
    </application>
   </appInfo>
  </encodingDesc>
  <workList>
   <work>
    <title />
   </work>
  </workList>
 </meiHead>
 <music>
  <body>
   <mdiv xml:id="mdiv-0000000722229810">
    <score xml:id="score-0000001458408129">
     <scoreDef xml:id="scoredef-0000000682402461">
      <staffGrp xml:id="staffgrp-0000001740796203">
       <staffDef xml:id="staffdef-0000000598469492" n="1" lines="5">
        <clef xml:id="clef-L2F1" shape="G" line="2" />
       </staffDef>
      </staffGrp>
     </scoreDef>
     <section xml:id="section-L1F1">
      <measure xml:id="measure-L1">
       <staff xml:id="staff-0000001242170464" n="1">
        <layer xml:id="layer-L1F1N1" n="1">
         <rest xml:id="rest-L4F1" dur="16" />
         <beam xml:id="beam-L5F1-L7F1">
          <note xml:id="note-L5F1" dur="16" oct="5" pname="c" accid.ges="n" />
          <note xml:id="note-L6F1" dur="16" oct="4" pname="e" accid.ges="n" />
          <note xml:id="note-L7F1" dur="16" oct="4" pname="f" accid.ges="n" />
         </beam>
         <tuplet xml:id="tuplet-L8F1-L10F1" num="3" numbase="2" bracket.visible="false" num.format="count">
          <beam xml:id="beam-L8F1-L10F1">
           <note xml:id="note-L8F1" dur="16" oct="5" pname="c" accid.ges="n" />
           <note xml:id="note-L9F1" dur="16" oct="4" pname="b" accid.ges="n" />
           <note xml:id="note-L10F1" dur="16" oct="4" pname="a" accid.ges="n" />
          </beam>
         </tuplet>
         <tuplet xml:id="tuplet-L11F1-L13F1" num="3" numbase="2" bracket.visible="false" num.format="count">
          <beam xml:id="beam-L11F1-L13F1">
           <note xml:id="note-L11F1" dur="16" oct="4" pname="g" accid.ges="n" />
           <note xml:id="note-L12F1" dur="16" oct="4" pname="f" accid.ges="n" />
           <note xml:id="note-L13F1" dur="16" oct="4" pname="e" accid.ges="n" />
          </beam>
         </tuplet>
        </layer>
       </staff>
      </measure>
     </section>
    </score>
   </mdiv>
  </body>
 </music>
</mei>
```

</details>

However, there is an additional typesetting rule that should be checked to break such a tie.  You must first look at the note/chord on either side of the current note/beam, and then if they both have the same direction, the choice would be to match the direction of both of the adjacent note/chord stem directions.   In this case the tie-breaker should be stems-up since the first and third beams are stem up.

Here is the discussion of such a case in Behind Bars, page 25:

<img width="588" alt="Screen Shot 2021-01-16 at 7 18 57 AM" src="https://user-images.githubusercontent.com/3487289/104815736-2e32f080-57cb-11eb-890c-e005c74d9539.png">

----------------------------

Unrelated, but interesting to think about (which I will post to the MEI issues):

This tie-breaking rule is relatively easy to implement, but there are other stem direction rules that would be more difficult to implement an algorithm for:

<img width="628" alt="Screen Shot 2021-01-16 at 7 21 15 AM" src="https://user-images.githubusercontent.com/3487289/104815765-6f2b0500-57cb-11eb-9c7a-5577c0f8ec08.png">

In other words, the four beams in this case should be considered to be contained within a single meta-beam, and this meta beam should  have stems down based on the basic stem direction algorithm.

So for such a case, creating a `<beamGrp>` element in MEI would be useful, since this sort of layout decision should not be left to an automatic decision.
